src/Pecee/Http/Input/InputHandler.php

```diff
@@ -62,33 +78,46 @@ public function __construct(Request $request)
      */
     public function parseInputs(): void
     {
+        $this->originalParams = [];
+        $this->get = [];
+        $this->originalPost = [];
+        $this->post = [];
+        $this->body_plain = '';
+        $this->originalBody = [];
+        $this->body = [];
+        $this->originalFile = [];
+        $this->file = [];
+
         /* Parse get requests */
         if (\count($_GET) !== 0) {
             $this->originalParams = $_GET;
             $this->get = $this->parseInputItem($this->originalParams);
         }
 
         /* Parse post requests */
-        $this->originalPost = $_POST;
-
-        if (\in_array($this->request->getMethod(), Request::$requestTypesPost, false) === true) {
-
-            $contents = file_get_contents('php://input');
+        if (\count($_POST) !== 0) {
+            $this->originalPost = $_POST;
+            $this->post = $this->parseInputItem($this->originalPost);
+        }
 
-            // Append any PHP-input json
-            if (strpos(trim($contents), '{') === 0) {
-                $post = json_decode($contents, true);
+        /* Get body */
+        $this->body_plain = file_get_contents('php://input');
 
-                if ($post !== false) {
-                    $this->originalPost += $post;
+        /* Parse body */
+        if (in_array($this->request->getMethod(), ['put', 'patch', 'delete', 'post'], false)) {
+            if(strpos($this->request->getContentType(), 'application/json') !== false){
+                $body = json_decode($this->body_plain, true);
+                if ($body !== false) {
+                    $this->originalBody = $body;
+                    $this->body = $this->parseInputItem($this->originalBody);
                 }
+            }else if(strpos($this->request->getContentType(), 'application/x-www-form-urlencoded') !== false){
+                parse_str(file_get_contents('php://input'), $body);
+                $this->originalBody = $body;
+                $this->body = $this->parseInputItem($this->originalBody);
```

Comment:
Hmm, I can test that but I thought that `PATCH` values for example only show up in the body and not the post.
I will test this and update

Reply:
`$this->originalBody` already contains `php://input` values.

Reply:
Yes you are right with that

Reply:
Wouldn't this do the trick `$this->post += $this->parseInputItem($this->originalBody)`? Why is separate property `body` needed? I'm pretty sure that `php://input` is raw `$_POST` values from other request-types. Maybe I'm mistaken...

Reply:
I will research a bit and write you then

Reply:
Okay,
application/x-www-form-urlencoded -> data in $_POST and php://input
multipart/form-data -> data in $_POST (and $_FILES)
application/json -> data in php://input

$_POST cannot parse JSON data. I will write more on it in the next comment after my push.

Reply:
Do you want to add json data to the post data? I think the name `POST` can lead to misunderstandings here because we mix request type with data store location. I can send a `PATCH` request using `application/x-www-form-urlencoded` and the data is stored in the `$_POST`.

We can still use the variable name `post` but we have to return the data for the following requests:
`POST, PUT, PATCH, DELETE`

What do you think?

Reply:
I think request-types like PATCH will be treated like POST on the webserver which is why the data can be found in `$_POST` (unless its special request-types like defined in php.net description). You can only have one request-type, so might as well merge the results of `php://input` to the post parameter. I think that's how it is currently.

Reply:
Okay, I have some ideas for that. You can see later in the push how I changed it.